jbi/actions/steps.py

```diff
@@ -1,89 +1,92 @@
 """
-Extended action that provides some additional features over the default:
-  * Updates the Jira assignee when the bug's assignee changes.
-  * Optionally updates the Jira status when the bug's resolution or status changes.
+Collection of reusable action steps.
 
-`init` is required; and requires at minimum the `jira_project_key` parameter. `status_map` is optional.
-
-`init` should return a __call__able
+Each step takes an `ActionContext` and a list of arbitrary parameters.
 """
+
 import logging
 
-from jbi import ActionResult, Operation
-from jbi.actions.default import (
-    JIRA_REQUIRED_PERMISSIONS as DEFAULT_JIRA_REQUIRED_PERMISSIONS,
-)
-from jbi.actions.default import (
-    add_link_to_bugzilla,
-    add_link_to_jira,
-    create_comment,
-    create_issue,
-    maybe_delete_duplicate,
-    update_issue,
-)
+from jbi import Operation
 from jbi.models import ActionContext
-from jbi.services import jira
+from jbi.services import bugzilla, jira
 
 logger = logging.getLogger(__name__)
 
 
-JIRA_REQUIRED_PERMISSIONS = DEFAULT_JIRA_REQUIRED_PERMISSIONS
+def create_comment(context: ActionContext, **parameters):
```

Comment:
I wonder if we should create a type for this function signature and enforce it somewhere. Probably not something for this PR, but something I'm thinking of.

Reply:
https://github.com/mozilla/jira-bugzilla-integration/issues/262